Initialize repo with ElasticLMv2

manhlx3006 opened the pull request.

vutt94 committed:
init foundry & copy KSElasticLMv2

danielpham765 committed:
SC-472 feat(*): Set up github actions lint, tests (#1)

* SC-472 feat(*): Set up github actions lint, tests

* SC-472 feat(config): Remove hardhat test

manhlx3006 commented on contracts/KSElasticLMV2.sol:
```diff
@@ -0,0 +1,1015 @@
+//SPDX-License-Identifier: MIT
```
Please try to use 2 spaces as convention

manhlx3006 commented on contracts/KSElasticLMV2.sol:
```diff
@@ -0,0 +1,1015 @@
+        //if phase not settled, update sumReward.
+        //if phase already settled then it's not needed since sumReward would be unchanged
+        if (block.timestamp > farms[fId].lastTouchedTime && !phase.isSettled)
+            _updateFarmSumRewardPerLiquidity(fId);
```
Should we move the check `block.timestamp > farms[fId].lastTouchedTime` into the `_updateFarmSumRewardPerLiquidity` function as it would be checked at any calls

vutt94 replied:
it would be check inside _calcSumRewardPerLiquidity.
In case block.timestamp == farms[fId].lastTouchedTime, _updateFarmSumRewardPerLiquidity would returns the lastest value.

manhlx3006 replied:
It's more consistent than we have to check it outside of each function that calls this function. Also we can ignore the result if we don't want to use

vutt94 replied:
updated, please help to check

manhlx3006 commented on contracts/KSElasticLMV2.sol:
```diff
@@ -0,0 +1,1015 @@
+        if (isInvalid) revert PositionNotEligible();
+
+        //calculate lastest farm sumReward
+        uint256[]
```
Code looks weird, should check the config for prettier or solhint

manhlx3006 commented on contracts/KSElasticLMV2.sol:
```diff
@@ -0,0 +1,1015 @@
+        }
+
+        //update farm lastTouchedTime, if passed endTime, update phase to settled
+        if (block.timestamp > lastTouchedTime)
```
local variable here?

vutt94 replied:
yeah it's local variable, loaded from the beginning of this func

manhlx3006 replied:
I mean should we use local var for block.timestamp as it is called multiple times

vutt94 replied:
yeah I can cache it to local variable but it is needed? I think gas consume would be the same

vutt94 committed:
re-format all

vutt94 committed:
updated: _updateFarmSumRewardPerLiquidity

vutt94 committed:
update readme
fix CI checks
add deploy script
add UT to fully coverage

vutt94 committed:
fix failed CI checks

vutt94 committed:
add UT for LMv2 with ElasticV2

vutt94 committed:
Merge pull request #3 from KyberNetwork/jensen_add-integration-test-with-ElasticV2

add UT for LMv2 with ElasticV2

vutt94 committed:
remove constructor param for farmingToken so it can be verify automatically

manhlx3006 committed:
Merge pull request #4 from KyberNetwork/jensen_remove-constructor-param-for-farmingToken

Remove constructor param for farmingToken

Zohar97 committed:
Add FairLaunchV3

Zohar97 committed:
Merge pull request #5 from KyberNetwork/add_FairLaunchV3

Add FairLaunchV3

vutt94 committed:
adding more test cases for LMv2

vutt94 committed:
update farmingToken:
- remove admin role
- remove whitelist, user can transfer freely

vutt94 committed:
update comments

vutt94 committed:
Merge pull request #6 from KyberNetwork/jensen_remove-admin-role-for-farmingToken

update farmingToken:

Zohar97 committed:
add network arb bsc

vutt94 committed:
add activate Range function

* Done:
- add activeRange function
- add UT for coverage
- add Integration cases

* fix naming for activateRange

Zohar97 committed:
update script and env on op and avax

vutt94 committed:
add 2 test cases for no rewards farm:
- rewardAmount = 0
- reward is dummy

vutt94 committed:
Add another test case for ETH

vutt94 committed:
fix issue farmingToken address zero

vutt94 committed:
Merge pull request #10 from KyberNetwork/jensen_add-UT-for-case-farm-without-rewards

add 2 test cases for no rewards farm:

vutt94 committed:
Merge branch 'develop' into jensen_fix-update-liquidity-not-checking-farmingToken

vutt94 committed:
add ut for case no token + no reward

vutt94 committed:
fix test withoutToken

manhlx3006 committed:
Merge pull request #11 from KyberNetwork/jensen_fix-update-liquidity-not-checking-farmingToken

Fix issue farmingToken address zero

Zohar97 committed:
add network config

vutt94 committed:
Updated: allow empty rewards farm (#12)

Zohar97 committed:
add config polygon zkevm

vutt94 committed:
lmv2 update: (#13)

* lmv2 update:
- store new lastTouchedTime even there is no rewards

* update comment for clearer understanding

vutt94 committed:
upgrade oz contracts to 4.3.2 (#14)

Zohar97 committed:
add base chain config

vutt94 committed:
Added flags param for removeLiquidity and claimFee: (#15)

* Added flags param for removeLiquidity and claimFee:
- isClaimFee (5th bit)
- isSyncFee (6th bit)
- isClaimReward (7th bit)
- isReceiveNative (8th bit)
For removeLiquidity: also claimReward if set isClaimReward to true
For claimFee: only call syncFee on posManager if set isSyncFee to true

* add indexed for fId in claimReward event

Zohar97 committed:
upgrade kyber swap role contract (#16)

Zohar97 committed:
Add safecheck for FairLaunchV3

Zohar97 committed:
add check to update and renew

manhlx3006 committed:
Merge pull request #17 from KyberNetwork/update-fairv3

Add safecheck for FairLaunchV3
